amaranth/ml/train.py

```diff
@@ -53,23 +55,34 @@ def main():
       high_calorie_threshold=amaranth.HIGH_CALORIE_THRESHOLD)
 
   # Do some preprocessing and calculations for encoding
-  corpus = calorie_data['description'].str.replace(',', '').str.lower()
+  calorie_data['description'] = calorie_data['description'].str.replace(
+      ',', '').str.lower()
+  corpus = calorie_data['description']
   vocab_size = lib.num_unique_words(corpus)
   tokenized_corpus = corpus.map(lambda desc: desc.split(' '))
   max_corpus_length = lib.max_sequence_length(tokenized_corpus)
 
-  # Create TextVectorization layer
-  vectorize_layer = keras.layers.experimental.preprocessing.TextVectorization(
-      max_tokens=vocab_size,
-      output_sequence_length=max_corpus_length,
-  )
+  # Encode 'description' into new column 'input'
+  calorie_data['input'] = calorie_data.apply(
```

Comment:
Make sure you're also implementing
- a max_token filter (which limits only the top X number of words to become tokens)
- a out-of-vocabulary (OOV) token, which you assign for every word that isn't part of your dictionary

Reply:
Very true, forgot about this since keras' TextVectorization layer did that for me. Fixed in https://github.com/googleinterns/amaranth/pull/35/commits/c5622901df47d6457b6cc7f5e720c58a48c6102a